### PR TITLE
Initial PR for 

### LINKED ISSUE #14: Support inheritance for child states for ui-router
Hello, very nice service!
What do you think about support for inheritance for child states in ui-router?
At the moment it doesn't work for 'title' for example, but does for 'description'. I think this is why you are resetting the current state of ngMeta object in readRouteMeta() function.
It would be really nice enhancement!

### PR DESCRIPTION
**Pull request by martyzz1**
Please Review, haven't been able to test this with ngRoute. only ui-router. 

Based on some feedback from the ui-router team here 
https://github.com/angular-ui/ui-router/issues/3045

This would help resolve #14

### CODE REVIEW

File: src/ngMeta.js [hunk 1]
```diff
@@ -167,9 +167,16 @@
         };
 
         var update = function(event, current) {
-          readRouteMeta(angular.copy(current.meta));
+          if (current.meta) {
```
Comment (vinaygopinath) [inline src/ngMeta.js]: Isn't this if-else chain equivalent to 

``` javascript
readRouteMeta(angular.copy(current.meta || current.data.meta));
```

File: README.md [hunk 2]
```diff
@@ -66,7 +66,55 @@ or download the file from [dist](https://github.com/vinaygopinath/ngMeta/tree/ma
       ...
     });
     ```
+**N.B. if you wish to take advantage of ui-router's nested views and data inheritence, then you should specify your "meta" config underneath the "data" property like this**
```
Comment (vinaygopinath) [inline README.md]: Would you mind moving this to a separate section after the "Dynamic meta tags" section? I wouldn't want to intimidate newcomers trying to set up ngMeta with anything more than the basic configuration needed to get up and running.

Also, given that it's backwards-compatible, I believe it's safe to move all the examples to use

``` javascript
data: {
  meta: {

  }
}
```

(I can deal with this after your PR is merged)

File: src/ngMeta.js [hunk 3]
```diff
@@ -276,8 +283,37 @@
...
+       *
+       * @returns {Object} data
+       */
+      this.mergeNestedStateData = function(state, parentDecoratorFn) {
+      // original data
+        var originalData = parentDecoratorFn(state) || {};
+
+        // create new merged meta
+        var parentMetaData = state.parent && state.parent.data && state.parent.data.meta;
+        var mergedMeta = angular.merge({}, parentMetaData, originalData.meta);
```
Comment (vinaygopinath) [inline src/ngMeta.js]: Shouldn't this statement be in the if block below? Calling angular.merge when both `parentMetaData` and `originalData.meta` are possibly null/undefined wouldn't serve a purpose anyway.

**Review by vinaygopinath — CHANGES_REQUESTED** [review]
Hey,

Thank you so much for offering to implement this feature, getting in touch with the ui-router folks and getting it done! Except for a few minor suggestions, I think the code looks good, and I'd be happy to put it through some tests, merge and release a new version.

**Comment by martyzz1** [comment]
hey no worries, thanks for the feedback... Done I believe

**Review by vinaygopinath — APPROVED** [review]

**Pull request merged**